### PR TITLE
promql: marshal sample pairs to JSON tuples.

### PR DESCRIPTION
**Pull request by fabxc**
@juliusv

### CODE REVIEW

File: web/api/legacy/query.go [hunk 1]
```diff
@@ -88,7 +90,35 @@ func (api *API) Query(w http.ResponseWriter, r *http.Request) {
...
-	httputil.RespondJSON(w, res.Value)
+	if vec, ok := res.Value.(promql.Vector); ok {
+		respondJSON(w, plainVec(vec))
+		return
+	}
+
+	respondJSON(w, res.Value)
+}
+
+// plainVec is an indirection that hides the original MarshalJSON methods
```
Comment (juliusv) [inline web/api/legacy/query.go]: methods -> method

**Comment by juliusv** [comment]
:+1: otherwise

**Comment by juliusv** [comment]
Actually, hold off. It also needs a change to the JS to understand the new vector result format.

**Comment by fabxc** [comment]
Adjusted the JSON, added string and scalar.

**Pull request merged**